### ISSUE TITLE
NoSuchMethodException URL.toURILenient

### ISSUE THREAD

**Issue by VladAbramov** [robolectric/robolectric#1322]
I have my own FileDownloader and in code I have:
```
URL url = new URL(urlLine.replaceAll(" ", "%20"));
URLConnection conn = url.openConnection();

if (!(conn instanceof HttpURLConnection)) {
    throw new IOException("Not an HTTP connection");
}
httpConn = (HttpURLConnection) conn;
httpConn.setAllowUserInteraction(false);
httpConn.setInstanceFollowRedirects(true);
httpConn.setConnectTimeout(Constants.FILE_DOWNLOAD_CONNECTION_TIMEOUT);
httpConn.setReadTimeout(Constants.FILE_DOWNLOAD_CONNECTION_TIMEOUT);
httpConn.setRequestMethod("GET");

httpConn.connect();
```
And after connect method invokation I have this exception. It's only happened when I run tests on robolectric (on android devices it's ok). It is hide method and maybe because of it this method does mot found in your framework.
```
Caused by: java.lang.NoSuchMethodError: java.net.URL.toURILenient()Ljava/net/URI;
	at libcore.net.http.HttpEngine.__constructor__(HttpEngine.java:195) ~[na:na]
	at libcore.net.http.HttpEngine.<init>(HttpEngine.java:188) ~[na:na]
	at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.<init>(HttpsURLConnectionImpl.java:421) ~[na:na]
	at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.<init>(HttpsURLConnectionImpl.java:400) ~[na:na]
	at libcore.net.http.HttpsURLConnectionImpl$HttpUrlConnectionDelegate.newHttpEngine(HttpsURLConnectionImpl.java:385) ~[na:na]
	at libcore.net.http.HttpURLConnectionImpl.initHttpEngine(HttpURLConnectionImpl.java:252) ~[na:na]
	at libcore.net.http.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:79) ~[na:na]
	at libcore.net.http.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:165) ~[na:na]
```

**Comment by jaredsburrows** [comment]
@erd @VladAbramov Try using Robolectric 3.0+.

    @Test
    public void test_shouldConnect() throws Exception {

        HttpURLConnection httpConn;
        URL url = new URL("http://google.com");
        URLConnection conn = url.openConnection();

        if (!(conn instanceof HttpURLConnection)) {
            throw new IOException("Not an HTTP connection");
        }
        httpConn = (HttpURLConnection) conn;
        httpConn.setAllowUserInteraction(false);
        httpConn.setInstanceFollowRedirects(true);
        httpConn.setConnectTimeout(5000);
        httpConn.setReadTimeout(5000);
        httpConn.setRequestMethod("GET");

        httpConn.connect();
    }

I get not exception here. I think we can close this.

**Comment by erd** [comment]
Sounds like it's fixed in 3.0.
